tests/_data/test_file_helpers.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) Lightly AG and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+#
+from __future__ import annotations
+
+from pathlib import Path
+
+import numpy as np
+import pytest
+from pytest_mock import MockerFixture
+
+from lightly_train._data import file_helpers
+
+from .. import helpers
+
+
+@pytest.mark.parametrize(
+    "extension, expected_backend",
+    [
+        (".jpg", "torch"),
+        (".jpeg", "torch"),
+        (".png", "torch"),
+        (".bmp", "pil"),
+        (".gif", "pil"),
+        (".tiff", "pil"),
+        (".webp", "pil"),
+    ],
+)
+def test_open_image_numpy(
+    tmp_path: Path, extension: str, expected_backend: str, mocker: MockerFixture
+) -> None:
+    image_path = tmp_path / f"image{extension}"
+    helpers.create_image(path=image_path, height=32, width=32)
+
+    torch_spy = mocker.spy(file_helpers, "_open_image_numpy__with_torch")
+    pil_spy = mocker.spy(file_helpers, "_open_image_numpy__with_pil")
+
+    result = file_helpers.open_image_numpy(image_path=image_path, mode="RGB")
+    assert isinstance(result, np.ndarray)
+    assert result.shape == (32, 32, 3)
+
+    if expected_backend == "torch":
+        torch_spy.assert_called_once()
+        pil_spy.assert_not_called()
+    else:
+        pil_spy.assert_called_once()
+        torch_spy.assert_not_called()
+
+
+def test_open_image_numpy__mask(tmp_path: Path) -> None:
+    image_path = tmp_path / "image.png"
+    helpers.create_image(path=image_path, height=32, width=32, mode="L")
+
+    result = file_helpers.open_image_numpy(image_path=image_path, mode="MASK")
+    assert isinstance(result, np.ndarray)
+    assert result.shape == (32, 32)
+    assert result.dtype == np.uint8
```

Comment:
Should we perhaps cover the test case where the `dtype` is not `np.unit8`? [io.decode_image](https://docs.pytorch.org/vision/main/generated/torchvision.io.decode_image.html#torchvision.io.decode_image) which is called by `io.read_image` says we could trigger that with a 16-bit png.

Reply:
I am also not sure - but could perhaps occur if someone creates a training set by just taking all the image material they have.